version: '3.8'

services:
  seq:
    image: datalust/seq:latest
    container_name: seq
    restart: unless-stopped
    ports:
      - 5341:80
    volumes:
      - /mnt/storage/seq-data:/data
    environment:
      - ACCEPT_EULA=Y
      - SEQ_FIRSTRUN_NOAUTHENTICATION=True
    networks:
      - traefik
    labels:
      - "traefik.enable=true"

      # Tell Traefik which network to use to reach the container
      - "traefik.docker.network=traefik"

      # HTTPS router
      - "traefik.http.routers.seq.rule=Host(`seq.cloudtorq.io`)"
      - "traefik.http.routers.seq.entrypoints=websecure"
      - "traefik.http.routers.seq.tls=true"
      - "traefik.http.routers.seq.tls.certresolver=letsencrypt"

      # Service: internal container port
      - "traefik.http.services.seq.loadbalancer.server.port=5341"

      # (Optional) HTTP -> HTTPS redirect router
      - "traefik.http.routers.seq-redirect.rule=Host(`seq.cloudtorq.io`)"
      - "traefik.http.routers.seq-redirect.entrypoints=web"
      - "traefik.http.routers.seq-redirect.middlewares=redirect-to-https"
      - "traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https"

networks:
  traefik:
    external: true